components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/dao/ActionManagementDAOImplTest.java

```diff
@@ -0,0 +1,341 @@
+            identityTenantUtil.when(()-> IdentityTenantUtil.getTenantId(anyString())).thenReturn(TENANT_ID);
+            SecretType secretType = mock(SecretType.class);
+            when(secretType.getId()).thenReturn("secretId");
+            when(secretManager.getSecretType(any())).thenReturn(secretType);
+        }
+        catch (Exception ignored){
+        }
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        identityTenantUtil.close();
+        identityDatabaseUtil.close();
+        identityUtil.close();
+    }
+
+    @AfterClass
+    public void wrapUp() throws Exception {
+        closeH2Database();
+    }
+
+    @Test
+    public void testAddAction() throws ActionMgtException {
+
+        Action action = getAction("120120");
+        Action result = daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        Assert.assertEquals(action.getName(),result.getName());
+        Assert.assertEquals(action.getDescription(),result.getDescription());
+        Assert.assertEquals(action.getType(),result.getType());
+        Assert.assertEquals(action.getStatus(),result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+
+    }
+
+    @Test
+    public void testAddActionWithoutOptional() throws ActionMgtException {
+
+        Action action = getActionWithoutOptionals("136136");
+        Action result = daoImpl.addAction(action.getType().getActionType(), action.getId(), action,TENANT_ID);
+        Assert.assertEquals(action.getName(),result.getName());
+        Assert.assertEquals(action.getType(),result.getType());
+        Assert.assertEquals(action.getStatus(),result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(),result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(), result.getEndpoint().getAuthentication().getType());
+
+    }
+
+    @Test
+    public void testGetActionsByActionType() throws ActionMgtException{
+
+        Action action = getAction("149149");
+        Assert.assertEquals(5,daoImpl.getActionsByActionType(action.getType().getActionType(),TENANT_ID).size());
+        loadMockDatabaseUtil();
+        daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        loadMockDatabaseUtil();
+        Assert.assertEquals(6,daoImpl.getActionsByActionType(action.getType().getActionType(),TENANT_ID).size());
+
+    }
+
+    @Test
+    public void testUpdateAction() throws ActionMgtException {
+
+        Action action = getAction("139139");
+        Action addedAction = daoImpl.addAction(action.getType().getActionType(), action.getId(), action, TENANT_ID);
+        loadMockDatabaseUtil();
+        Action updatedAction = new Action.ActionResponseBuilder()
```

Comment:
```suggestion
        Action updatingAction = new Action.ActionRequestBuilder()
```

Use ActionRequestBuilder() to build the objects use to create and update actions